source/_integrations/zha.markdown

```diff
@@ -446,15 +446,27 @@ ZHA does not currently support devices that can only use the ZSE ("Zigbee Smart
+
+#### How to add support for new and unsupported devices
+
+If you have already followed the recommended troubleshooting steps and your Zigbee device pairs/joins successfully with the ZHA integration but still does not show any entities or not all of the expected entities, even after trying to re-pair/re-join it several times, you may need a custom device handler. This handler will have exception code to work around specific device deviations or unique features and functions.
+
+For devices that do not follow the standard clusters and attributes provided in the CSA's ZCL (Zigbee Cluster Library) specifications, the ZHA integration relies on a project called "[ZHA Device Handlers (also known as "zha-quirk")](https://github.com/zigpy/zha-device-handlers)". It contains device-specific Python scripts, referred to in ZHA as "quirks", that can resolve compliance and interoperability issues by implementing on-the-fly conversion/translation of custom Zigbee configurations and parameters or implementing manufacturer-specific features and functions for specific devices.
+
+People familiar with other Zigbee gateway solutions for home automation may know similar concepts of using custom Zigbee device handlers/converters for non-standard devices. For example, [Zigbee2MQTT (and IoBroker) uses zigbee-herdsman converters](https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html) and [SmartThings Classics (Legacy) platform has Hub Connected Device Handlers](https://developer.smartthings.com/docs/devices/hub-connected/legacy).
+
+If you are not able to develop such a custom "quirk" Python script yourself with the custom handler/converter/parser/translator code needed for your specific device and add it to the "ZHA Device Handlers" repository/library, you can submit a "device support request" as a new issue to the [ZHA Device Handlers project repository on GitHub](https://github.com/zigpy/zha-device-handlers/issues). To do so, sign up/in on GitHub and click "New issue" and follow the instructions. New device support requests requires at minimum that the device owner supply the device signature + diagnostic information. The device owner may also need to actively assist in further testing or provide additional information that volunteering developers might to create custom ZHA Device Handlers for your devices.
+
+Note that submitting a new "device support request" does not guarantee that someone else will develop a custom "quirk" for ZHA, as it relies on volunteering developers contributing new ZHA Device Handlers to the project. However, without "device support requests" being submitted for deviating devices to the ZHA Device Handlers project, it is likely that the ZHA integration developers will not know that the specific Zigbee device is not working correctly in ZHA unless they happen to have bought it themselves by chance.
```

Comment:
```suggestion
Note that submitting a new "device support request" does not guarantee that someone else will develop a custom "quirk" for ZHA. The project relies on volunteering developers. However, without "device support requests", the developers may not be aware that your specific Zigbee device is not working correctly in ZHA. 
```

Reply:
In some instances, I took the liberty of heavy editing. I think sometimes, less is more. At least for me, this level of detail can be overwhelming. For example, when near-synonyms (such as features and functionality) were used, I removed one because I find it easier to read. At the same time, I don't think much is lost when removing one or the other. Let me know what you think.